Introduce manifest and support files for dev.restfox.Restfox

x80486 opened the pull request:
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.

Restfox is a REST client application that allows users to test and interact with RESTful APIs. It provides a user-friendly interface for making HTTP requests, viewing responses, and managing API endpoints. Restfox is designed to help developers and testers streamline their API development and testing processes. Additionally, Restfox offers plugin support to help developers modify request data before a request is sent, adjust response data after a response is received, set environment variables using the response data, and perform testing.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/flawiddsouza/Restfox/issues/71


<!-- ⚠️  Don't modify anything below this line ⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

hfiguiere commented on dev.restfox.Restfox.yaml:
```diff
@@ -0,0 +1,40 @@
+          - export TMPDIR="${XDG_RUNTIME_DIR}/app/dev.restfox.Restfox"
+          - exec zypak-wrapper /app/lib/restfox/Restfox "$@"
+        dest-filename: entrypoint.sh
+rename-desktop-file: restfox.desktop
```
these last line are misplaced and should be in the top section.

x80486 replied:
I use `eslint-plugin-yml` for YAML, and it sorts they keys alphabetically by default.

hfiguiere replied:
then don't. I mentionned it yesterday on Matrix they were misplaced. Does this sort the javascript line? I don't think so.

hfiguiere commented on dev.restfox.Restfox.yaml:
```diff
@@ -0,0 +1,40 @@
+      - type: file
+        dest-filename: restfox.deb
+        sha256: 22e5434b69eff39d1c42ef503801b3c4aa578a4eeb26389ca20163ab32520585
+        url: https://github.com/flawiddsouza/Restfox/releases/download/v0.29.0/restfox_0.29.0_amd64.deb
```
why not building from source?

bbhtt commented on dev.restfox.Restfox.yaml:
```diff
@@ -0,0 +1,40 @@
+      - ar -o -x restfox.deb
+      - tar --extract --file data.tar.gz
+      - cp --preserve --recursive usr/* /app
+      - desktop-file-edit --set-key Exec --set-value "/app/bin/restfox %U" /app/share/applications/restfox.desktop
```
```suggestion
      - desktop-file-edit --set-key Exec --set-value "restfox %U" /app/share/applications/restfox.desktop
```

x80486 replied:
If you don't mind, I prefer to leave it with the path, because there could be at any point a few `restfox` in the path — since this tends to change as well with time.

The way it's right now, there won't be any mistakes — if something changes, it's this one thing.

hfiguiere replied:
how could there be? It's inside of the sandbox. and its content doesn't vary without rebuild.

x80486 replied:
The suggested change is to take a file named, `restfox`, resolved dynamically based on the current contents pf the `$PATH` variable — that's how I understand this.

What I eas explaining was that the `$PATH` variable can change over time as runtimes change/evolve; the content of the package may also be different, and also the manifest will change with time. As a result, there could be multiple versions of `restfox` that are accessible at any given moment.

However, the likelihood of this issue to happen is quite low...so I won't sweat it and adjust it.

bbhtt commented on dev.restfox.Restfox.yaml:
```diff
@@ -0,0 +1,40 @@
+      - desktop-file-edit --set-key Exec --set-value "/app/bin/restfox %U" /app/share/applications/restfox.desktop
+      - install -D --mode 0644 /app/lib/restfox/resources/app/ui/favicon.svg /app/share/icons/hicolor/scalable/apps/dev.restfox.Restfox.svg
+      - install -D --mode 0644 --target-directory /app/share/metainfo/ dev.restfox.Restfox.metainfo.xml
+      - install -D --mode 0755 entrypoint.sh /app/bin/restfox # Overwrites sysmlink ¯\_(ツ)_/¯
```
```suggestion
      - install -D --mode 0755 entrypoint.sh /app/bin/restfox
```

bbhtt commented on dev.restfox.Restfox.metainfo.xml:
```diff
@@ -0,0 +1,35 @@
+  <launchable type="desktop-id">dev.restfox.Restfox.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/flawiddsouza/Restfox/refs/heads/main/screenshots/1.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://raw.githubusercontent.com/flawiddsouza/Restfox/refs/heads/main/screenshots/2.png</image>
+    </screenshot>
```
screenshots should from a commit not `refs/heads/main`

bbhtt commented on dev.restfox.Restfox.yaml:
```diff
@@ -0,0 +1,40 @@
+command: restfox
+finish-args:
+  - --device=dri
+  - --filesystem=home
```
why is home needed?

x80486 replied:
Files/Collections need to be saved/read from the filesystem. It's difficult to lock down users to certain locations, so to maintain the same experience with the original release, this is needed at the moment.

This application behaves similar to Postman and Insomnia.

x80486 commented on flathub.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "only-arches": [
+    "x86_64"
```
Later on, if building from sources work, ARM could be added as well.

x80486 commented:
bot, build dev.restfox.Restfox

bbhtt commented on dev.restfox.Restfox.metainfo.xml:
was this submitted upstream?

x80486 replied:
I did not. I will try to make a list of changes that will streamline the Flatpak release workflow in the Restfox's issue tracker.

Nevertheless, I will summon @flawiddsouza because he is the owner/maintainer of Restfox, so he's aware of the changes — I will be a second hand if/when Restfox is finally accepted in Flathub.

flawiddsouza replied:
Hi. Let me know what needs to be done and I'll do it as soon as I can.

bbhtt replied:
> Hi. Let me know what needs to be done and I'll do it as soon as I can.

You want to take in the metainfo file from here and ship it with any package you distribute. It's a cross-desktop Freedesktop standard that can be used by debs/rmps etc. as well.

x80486 commented:
bot, build dev.restfox.Restfox

x80486 commented:
bot, build dev.restfox.Restfox

x80486 commented:
bot, build dev.restfox.Restfox

x80486 commented:
@hfiguiere and/or @bbhtt, is there anything else pending for this application?

The status reads as `awaiting-changes`, but I've addressed all comments.

hfiguiere commented:
then have some patience instead.

bbhtt commented:
> @hfiguiere and/or @bbhtt, is there anything else pending for this application?
> 
> The status reads as `awaiting-changes`, but I've addressed all comments.

I didn't see any movement about https://github.com/flathub/flathub/pull/5746#discussion_r1814207482

flawiddsouza commented:
> I didn't see any movement about [#5746 (comment)](https://github.com/flathub/flathub/pull/5746#discussion_r1814207482)

Hi, I've sadly not been able to find any documentation on how to package the metainfo file with an electron application. I found this issue https://github.com/electron-userland/electron-builder/issues/1993 but I don't think there's been any update. If you have any helpful documentation, please point me to it.

bbhtt commented:
Can you please take in the copy? It can be packaged later when electron builder supports it.

flawiddsouza commented:
Done: https://github.com/flawiddsouza/Restfox/commit/780afdd99e09e2092e4664199ba59cb7c0847cc2.

bbhtt commented:
/merge
